mavros_extras/src/plugins/odom.cpp

```diff
@@ -198,35 +110,26 @@ class OdometryPlugin : public plugin::PluginBase {
 	 * @brief Handle ODOMETRY MAVlink message.
 	 *
 	 * Message specification: https://mavlink.io/en/messages/common.html#ODOMETRY
+	 *
+	 * Callback to odometry that is sent from FCU by mavlink_messages. It expects
+	 * the parent frame of the msg to be "local_origin_ned" and the child frame to be "fcu_frd".
+	 * The odometry message gets rotated such that parent frame and child frame are the desired
+	 * frames set in the rosparams "odometry/in/fcu_odom_*_id_des" (set in px4_config.yaml). If these frames are not part
+	 * of the usual tf tree, the user is supposed to add these frames to the existing tf tree.
```

Comment:
"If these frames are not part of the usual tf tree, the user is supposed to add these frames to the existing tf tree."

This is unclear. I don't see why it needs to be added to the tree manually, and how this is supposed to work.

Reply:
The external odometry sensor or mocap system will most likely introduce two frames. A reference frame with a for us unknown orientation. And the same for the link on the body. The user of the plugin has to specifiy how these frames are related to the odom reference frame as well as to the base_link. With this information we will be able to roatete the odometry message in the frames we will expect on the px4 side.